service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -536,31 +506,16 @@ public void removeResourceRole(
    * called otherwise.
    *
    * <p>This call to Sam is made as the WSM SA, as users do not have permission to directly modify
-   * IAM on resources. This method still requires user credentials to validate as a safeguard, but
-   * they are not used in the role removal call.
+   * IAM on resources.
    *
    * @param resource The resource to restore a role to
-   * @param userRequest User credentials. These are not used for the call to Sam, but must belong to
-   *     a workspace owner to ensure the WSM SA is being used on a user's behalf correctly.
    * @param role The role to restore
    * @param email Email identifier of the user whose role is being restored.
    */
   @Traced
   public void restoreResourceRole(
-      ControlledResource resource,
-      AuthenticatedUserRequest userRequest,
-      ControlledResourceIamRole role,
-      String email)
+      ControlledResource resource, ControlledResourceIamRole role, String email)
       throws InterruptedException {
-    // Validate that the provided user credentials can modify the owners of the resource's
-    // workspace.
-    // Although the Sam call to revoke a resource role must use WSM SA credentials instead, this
-    // is a safeguard against accidentally invoking these credentials for unauthorized users.
-    checkAuthz(
```

Comment:
The Sam call here is made as WSM's SA, but at the controller layer before we get to this method we've already verified that the user has the required permission so we don't need the additional check here.

We could keep a duplicate check here in case of developer mistake, but part of the driver for this refactor was giving us confidence that we don't need to because it's easier to see every endpoint is doing authorization checks by inspection.